src/game/utils/batch/report.rs

```diff
@@ -0,0 +1,296 @@
+use std::{
+    fmt::Display,
+    fs::{create_dir, File},
+    io::Write,
+};
+
+use crate::{compute_average, game::utils::flop_helper::flop_to_string, Action, PostFlopGame};
+
+/// Returns the player's equity, EV, and EQR (in that order)
+/// Equity and EQR are float values from 0-100 (percentages)
+fn get_player_stats(game: &PostFlopGame, player: usize) -> (f32, f32, f32) {
+    let equity = game.equity(player);
+    let ev = game.expected_values(player);
+    let weights = game.normalized_weights(player);
+    let average_equity = compute_average(&equity, weights);
+    let average_ev = compute_average(&ev, weights);
+    (
+        100.0 * average_equity,
+        average_ev,
+        100.0 * average_ev / (average_equity * game.pot() as f32),
+    )
+}
+
+fn get_action_percentages(game: &PostFlopGame) -> Vec<f32> {
+    let player = game.current_player();
+    let cards = game.private_cards(player);
+    let strategy = game.strategy();
+    let actions = game.available_actions();
+    let weights = game.normalized_weights(player);
+
+    (0..actions.len())
+        .map(|i| compute_average(&strategy[i * cards.len()..(i + 1) * cards.len()], weights))
+        .collect()
+}
+
+// TODO is there any better way to do this?
+// I would rather not have to replay histories here b/c it is complicated and possibly slow
+// NOTE: Since this mutates the game, it un-caches weights
+fn get_action_evs(game: &mut PostFlopGame) -> Vec<f32> {
+    let actions = game.available_actions();
+    (0..actions.len())
+        .map(|action_index| {
+            let player = game.current_player();
+            let history = game.history().to_owned();
+            game.play(action_index);
+            game.cache_normalized_weights();
+
+            let evs = game.expected_values(player);
+            let average_ev = evs.iter().sum::<f32>() / evs.len() as f32;
+
+            game.back_to_root();
+            game.apply_history(history.as_slice());
+
+            average_ev
+        })
+        .collect()
+}
+
+fn is_terminating_action(action: Action, player: usize) -> bool {
+    match action {
+        Action::AllIn(_) => false,
+        Action::Bet(_) => false,
+        Action::Raise(_) => false,
+        // Check is terminating iff current player is IP
+        Action::Check => player == 1,
```

Comment:
`Action::Call` isn't always terminating. I don't think we can infer `is_terminating` from just the Action

Reply:
This was named poorly (meant to be `is_street_terminating_action`), but we no longer need it now that we're handling turn and river reports more generally.